service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java

```diff
@@ -0,0 +1,129 @@
+
+        messageContext = new MessageContext(randomIdGeneratorService);
+
+        DiaryPlanStatementMapper diaryPlanStatementMapper = new DiaryPlanStatementMapper(messageContext);
+        NarrativeStatementMapper narrativeStatementMapper = new NarrativeStatementMapper(messageContext);
+        ObservationStatementMapper observationStatementMapper = new ObservationStatementMapper(messageContext);
+        ImmunizationObservationStatementMapper immunizationObservationStatementMapper =
+            new ImmunizationObservationStatementMapper(messageContext);
+        ConditionLinkSetMapper conditionLinkSetMapper = new ConditionLinkSetMapper(messageContext, randomIdGeneratorService);
+
+        encounterComponentsMapper = new EncounterComponentsMapper(
+            messageContext,
+            diaryPlanStatementMapper,
+            narrativeStatementMapper,
+            observationStatementMapper,
+            immunizationObservationStatementMapper,
+            conditionLinkSetMapper);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        messageContext.resetMessageContext();
+    }
+
+    @ParameterizedTest
+    @MethodSource("testData")
+    public void When_MappingEncounterComponents_Expect_ResourceMapped(String inputJsonPath, String expectedXmlPath) throws IOException {
+        String expectedXml = ResourceTestFileUtils.getFileContent(expectedXmlPath);
+
+        String inputJson = ResourceTestFileUtils.getFileContent(inputJsonPath);
+        Bundle bundle = new FhirParseService().parseResource(inputJson, Bundle.class);
+        messageContext.initialize(bundle);
+
+        Optional<Encounter> encounter = extractEncounter(bundle);
+        assertThat(encounter.isPresent()).isTrue();
+
+        String mappedXml = encounterComponentsMapper.mapComponents(encounter.get());
+        assertThat(mappedXml).isEqualToIgnoringWhitespace(expectedXml);
+    }
+
+    @ParameterizedTest
+    @MethodSource("emptyResult")
+    public void When_MappingEncounterComponents_Expect_NoResourceMapped(String inputJsonPath) throws IOException {
+        String inputJson = ResourceTestFileUtils.getFileContent(inputJsonPath);
+        Bundle bundle = new FhirParseService().parseResource(inputJson, Bundle.class);
+        messageContext.initialize(bundle);
+
+        Optional<Encounter> encounter = extractEncounter(bundle);
+        assertThat(encounter.isPresent()).isTrue();
+
+        String mappedXml = encounterComponentsMapper.mapComponents(encounter.get());
+        assertThat(mappedXml).isEmpty();
+    }
+
+    private Optional<Encounter> extractEncounter(Bundle bundle) {
+        return bundle.getEntry().stream()
+            .map(Bundle.BundleEntryComponent::getResource)
+            .filter(resource -> ResourceType.Encounter.equals(resource.getResourceType()))
+            .map(resource -> (Encounter) resource)
+            .findFirst();
+    }
+
+    private static Stream<Arguments> testData() {
```

Comment:
Is there a point of making a parametrized test for only one set of test data?

Reply:
right, I should change that.